app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java

```diff
@@ -190,23 +193,36 @@ private void convertToPdf(String mFilename) {
         String mStorePath = mSharedPreferences.getString(STORAGE_LOCATION,
                 getDefaultStorageLocation());
         String mPath = mStorePath + mFilename + mActivity.getString(R.string.pdf_ext);
-        Workbook workbook;
-        try {
-            workbook = new Workbook(mRealPath);
-            workbook.save(mPath, FileFormatType.PDF);
-
-            getSnackbarwithAction(mActivity, R.string.snackbar_pdfCreated)
-                    .setAction(R.string.snackbar_viewAction, v -> mFileUtils.openFile(mPath)).show();
-            mTextView.setVisibility(View.GONE);
-
-        } catch (Exception e) {
-            e.printStackTrace();
-            showSnackbar(mActivity, R.string.error_occurred);
-        } finally {
-            mMorphButtonUtility.morphToGrey(mCreateExcelPdf, mMorphButtonUtility.integer());
-            mCreateExcelPdf.setEnabled(false);
-            mExcelFileUri = null;
-        }
+        mMaterialDialog = createAnimationDialog(mActivity);
+        mMaterialDialog.show();
+
+        new AsyncTask() {
```

Comment:
I think it will be better if you do create a new class by extending `AsyncTask`. And call that here.
What do you think @Swati4star?

Reply:
Alright @vyankatesh24 Will be doing that.

Reply:
@vyankatesh24 But then it will be difficult to handle the `onPostExecute tasks like setting the morphingButton to false, nullifying the uri, paths, the other animators, etc.

Reply:
> @vyankatesh24 But then it will be difficult to handle the `onPostExecute` tasks like setting the morphingButton to false, nullifying the uri, paths, the other animators, toggling the Dialog animation, etc.

@jddeep 
I think there will not be any problem if you create a class in the same file. But wait let @Swati4star reply.

Reply:
There wont be any difficulty in handling the postexecution too. You can look into ImageToPdfFragment.java also. We created a new interface `OnPDFCreatedInterface` and we send out an instance. Look at that workflow, if you find some difficulty there, let me know.

It'll be better to move the code to a different class only.